pkg/sentry/fsimpl/fuse/fusefs.go

```diff
@@ -456,6 +456,34 @@ func (i *inode) NewDir(ctx context.Context, name string, opts vfs.MkdirOptions)
 	return i.newEntry(kernelTask, req, name, linux.S_IFDIR, false)
 }
 
+// RmDir implements kernfs.Inode.RmDir.
+func (i *inode) RmDir(ctx context.Context, name string, child *vfs.Dentry) error {
+	fusefs := i.fs
+	task, creds := kernel.TaskFromContext(ctx), auth.CredentialsFromContext(ctx)
+
+	in := linux.FUSERmDirIn{Name: name}
+	req, err := fusefs.conn.NewRequest(creds, uint32(task.ThreadID()), i.NodeID, linux.FUSE_RMDIR, &in)
+	if err != nil {
+		return err
+	}
+
+	res, err := i.fs.conn.Call(task, req)
+	if err != nil {
+		return err
+	}
+	if err := res.Error(); err != nil {
+		return err
+	}
+
+	// TODO(Before merging): When creating new nodes, should we add nodes to the ordered children?
```

Comment:
How does existing implementation handle this? Should it happen after a response with no error? If so, check the `asyncCallback()` out in #3548 (formally known as `noReceiverAction()`)